### PR TITLE
Add relation async flag

### PR DESCRIPTION
**Pull request by MayTetro1**
Add usage monitor relation Async flag.
Installation refactor + bug fixes.

### CODE REVIEW

File: server-side/relations.service.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        await this.createAduitLogSettingsRelation();
+        await this.createNewAduitLogSettingsRelation();
+    }
+
+    async createUsageMonitorRelations(){
+        await this.upsertTransactionActivitiesRelation();
+        await this.upsertComputingTimeRelation();
+    }
+
+    async createPageComponentRelation() {
```
Comment (roiaharon-pepperi) [inline server-side/relations.service.ts:42]: the name of the function is incorrect, you're not creating a page component relation, but rather addon block relation.

File: server-side/relations.service.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            AddonUUID: this.client.AddonUUID,
+            AddonRelativeURL: this.filename,
+            ComponentName: `AddonComponent`,
+            ModuleName: `AddonModule`,
+            ElementsModule: 'WebComponents',
+            ElementName: `settings-element-${this.client.AddonUUID}`,
+        });
+    } 
+
+    async createNewAduitLogSettingsRelation() {
```
Comment (roiaharon-pepperi) [inline server-side/relations.service.ts:76]: what's the difference between this function and the previous pageComponent function?

File: server-side/relations.service.ts
Comment (roiaharon-pepperi) [inline server-side/relations.service.ts:1]: if you're done this refactor, then do it all the way mark the relevant functions as private..
although I would create a single function to create all relations.

File: server-side/installation.ts [hunk 3]
```diff
@@ -58,138 +61,28 @@ export async function uninstall(client: Client, request: Request): Promise<any>
...
-        console.log(`upsertComputingTimeRelation: ${ex}`);
-        throw new Error("Error in upsertComputingTimeRelation: " + ex);
+    const relationService = new RelationsService(client);
+    try {
+        const subscription = await service.papiClient.notification.subscriptions.find({ where: `Name='AuditDataLog'` });
+        if (subscription.length === 0) {
+            await insertSubscription(client, service);
+        }
+        
+        if (Semver.lte(request.body.FromVersion, '1.0.0')) { // for versions below 1.1.0
```
Comment (roiaharon-pepperi) [inline server-side/installation.ts:71]: I think that this is irrelevant anymore, there are no distributors below version 1.0

**Review by roiaharon-pepperi — CHANGES_REQUESTED** [review]

File: server-side/installation.ts (hunk 3 above)
Comment (MayTetro1) [reply server-side/installation.ts:71]: there are some distributors below that version

**Review by roiaharon-pepperi — APPROVED** [review]

**Pull request merged**